lib/ansible/modules/network/avi/avi_api_session.py

```diff
@@ -0,0 +1,192 @@
+
+short_description: Avi API Module.
+description:
+    - This module can be used for calling any resources defined in Avi REST API. This module is useful for invoking HTTP Patch methods and accessing resources that do not have an REST object associated with them.
+version_added: 2.3
+requirements: [ avisdk ]
+options:
+    http_method:
+        description:
+            - Allowed HTTP methods for RESTful services and are supported by Avi Controller.
+        choices: ["get", "put", "post", "patch", "delete"]
+        required: true
+    data:
+        description:
+            - HTTP body in YAML format.
+    data_json:
+        description:
+            - HTTP body in JSON format.
+    params:
+        description:
+            - Query parameters passed to the HTTP API.
+    path:
+        description:
+            - Path for Avi API resource. For example, C(path: virtualservice) will translate to C(api/virtualserivce).
+    timeout:
+        description:
+            - Timeout for Avi API calls.
+extends_documentation_fragment:
+    - avi
+'''
+
+EXAMPLES = '''
+# Get Pool Information using avi_api_session
+- avi_api_session:
+    # get pool information
+    controller: "{{ controller }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    http_method: get
+    path: pool
+    params:
+      name: "{{ pool_name }}"
+  register: pool_results
+
+# Patch Pool with list of servers
+  - avi_api_session:
+      controller: "{{ controller }}"
+      username: "{{ username }}"
+      password: "{{ password }}"
+      http_method: patch
+      path: "{{ pool_path }}"
+      data:
+        add:
+          servers:
+            - ip:
+                addr: 10.10.10.10
+                type: V4
+            - ip:
+                addr: 20.20.20.20
+                type: V4
+'''
+
+
+RETURN = '''
+obj:
+    description: Avi REST resource
+    returned: success, changed
+    type: dict
+'''
+
+
+def main():
+    argument_specs = dict(
+        http_method=dict(required=True,
+                         choices=['get', 'put', 'post', 'patch',
+                                  'delete']),
+        path=dict(required=True),
+        params=dict(type='dict'),
+        data=dict(type='dict'),
+        data_json=dict(type='str'),
+        timeout=dict(default=60)
+    )
+    argument_specs.update(avi_common_argument_spec())
+    module = AnsibleModule(argument_spec=argument_specs)
+
+    if not HAS_AVI:
+        return module.fail_json(msg=(
+            'Avi python API SDK (avisdk) is not installed. '
+            'For more details visit https://github.com/avinetworks/sdk.'))
+    tenant_uuid = module.params.get('tenant_uuid', None)
+    api = ApiSession.get_session(
+        module.params['controller'], module.params['username'],
+        module.params['password'], tenant=module.params['tenant'],
+        tenant_uuid=tenant_uuid)
+
+    tenant = module.params.get('tenant', '')
+    timeout = int(module.params.get('timeout', 60))
```

Comment:
You've specified the default here and in `argument_specs` which doesn't seem right